File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def get_product(self, n: int) -> int:
        x = n
        product = 1
        while x > 0:
            digit = x%10
            product *= digit
            x //= 10
        return product

    def smallestNumber(self, n: int, t: int) -> int:
        for i in range(n, 101):
            product = self.get_product(i)
            if product % t == 0:
                return i
        
        return -1
        

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    long long get_product(int& n){
        int x = n;
        long long product = 1;
        while(x > 0){
            int digit = x % 10;
            product *= digit;
            x = x/10;
        }

        return product;
    }
    int smallestNumber(int n, int t) {
        for(int i = n; i <= 100; i++){
            long long product = get_product(i);
            if(product % t == 0){
                return i;
            }
        }

        return -1;
    }
};